### PR TITLE
Add a convenient script to create internal PRs

### PR DESCRIPTION
**Pull request by npaun**
Now you can use `just internal-pr` to create or update your internal PR as needed.

### CODE REVIEW

File: tools/unix/create-internal-pr.sh [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	git checkout master 
+	git checkout -b $SYNC_BRANCH_NAME
+fi
+
+git -C deps/workerd fetch origin $BRANCH_NAME
+git -C deps/workerd checkout origin/$BRANCH_NAME
+git add deps/workerd
+
+if [ $BRANCH_EXISTS -eq 1 ]; then
+	git commit --amend
```
Comment (anonrig) [inline tools/unix/create-internal-pr.sh:32]:
```suggestion
	git commit --amend --no-verify
```

File: tools/unix/create-internal-pr.sh [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+	git push --force origin HEAD
```
Comment (anonrig) [inline tools/unix/create-internal-pr.sh:33]:
```suggestion
	git push --force --no-verify origin HEAD
```

File: tools/unix/create-internal-pr.sh [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+else
+	git commit -m "Sync with $BRANCH_NAME"
```
Comment (anonrig) [inline tools/unix/create-internal-pr.sh:35]:
```suggestion
	git commit --no-verify -m "Sync with $BRANCH_NAME"
```

File: tools/unix/create-internal-pr.sh [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+	BRANCH_EXISTS=1
+else 
+	BRANCH_EXISTS=0
+fi
+
+
+if [ $BRANCH_EXISTS -eq 1 ]; then
+	git checkout $SYNC_BRANCH_NAME
+else
+	git checkout master
```
Comment (anonrig) [inline tools/unix/create-internal-pr.sh:23]: You also need to make sure master branch is same as origin master in here.

File: tools/unix/create-internal-pr.sh [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+
+if [ $BRANCH_EXISTS -eq 1 ]; then
+	git checkout $SYNC_BRANCH_NAME
+else
+	git checkout master 
+	git checkout -b $SYNC_BRANCH_NAME
+fi
+
+git -C deps/workerd fetch origin $BRANCH_NAME
```
Comment (anonrig) [inline tools/unix/create-internal-pr.sh:27]: You can rely on `gh` cli tool here. I mostly do `gh pr checkout PR_ID` whenever I navigate to deps/workerd.

Comment (npaun) [reply tools/unix/create-internal-pr.sh:27]: Yeah I use it too but I wanted the script to work without it.

File: tools/unix/create-internal-pr.sh (hunk 4 above)
Comment (npaun) [reply tools/unix/create-internal-pr.sh:23]: Shouldn't line 11 have taken care of this? (It does an unconditional `git fetch origin master`

**Review by anonrig — APPROVED** [review]

**Pull request merged**